modules/wolfvision/eye14.rb

```diff
@@ -0,0 +1,167 @@
+module Wolfvision; end
+
+# Documentation: https://www.wolfvision.com/wolf/protocol_command_wolfvision/protocol/commands_eye-14.pdf
+
+class Wolfvision::Eye14
+    include ::Orchestrator::Constants
+    include ::Orchestrator::Transcoder
+
+    # Discovery Information
+    tcp_port 50915 # Need to go through an RS232 gatway
+    descriptive_name 'WolfVision EYE-14'
+    generic_name :Camera
+
+    # Communication settings
+    tokenize indicator: /\x00|\x01|/, callback: :check_length
+    delay between_sends: 150
+
+    def on_load
+        self[:zoom_max] = 3923
+        self[:iris_max] = 4094
+        self[:zoom_min] = self[:iris_min] = 0
+        on_update
+    end
+
+    def on_update
+    end
+
+    def on_unload
+    end
+
+    def connected
+        schedule.every('60s') do
+            logger.debug "-- Polling Sony Camera"
+            power? do
+                if self[:power] == On
+                    zoom?
+                    iris?
+                    autofocus?
+                end
+            end
+        end
+    end
+
+    def disconnected
+        # Disconnected will be called before connect if initial connect fails
+        schedule.clear
+    end
+
+    def power(state)
+        target = is_affirmative?(state)
+        self[:power_target] = target
+
+        # Execute command
+        logger.debug { "Target = #{target} and self[:power] = #{self[:power]}" }
+        if target == On && self[:power] != On
+            send_cmd("\x30\x01\x01", name: :power_cmd)
+        elsif target == Off && self[:power] != Off
+            send_cmd("\x30\x01\x00", name: :power_cmd)
+        end
+    end
+
+    # uses only optical zoom
+    def zoom(position)
+        val = in_range(position, self[:zoom_max], self[:zoom_min])
+        self[:zoom_target] = val
+        val = sprintf("%04X", val)
```

Comment:
Instead of `sprintf` can you use the `format` function - otherwise exactly the same as sprintf